src/Umbraco.Cms.ManagementApi/Controllers/ExamineManagement/RebuildIndexExamineManagementController.cs

```diff
@@ -0,0 +1,120 @@
+﻿using Examine;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Umbraco.Cms.Core.Cache;
+using Umbraco.Cms.Infrastructure.Examine;
+using Umbraco.Cms.ManagementApi.Services;
+
+namespace Umbraco.Cms.ManagementApi.Controllers.ExamineManagement;
+
+[ApiVersion("1.0")]
+public class RebuildIndexExamineManagementController : ExamineManagementControllerBase
+{
+    private readonly IExamineManagerService _examineManagerService;
+    private readonly ILogger<RebuildIndexExamineManagementController> _logger;
+    private readonly IIndexRebuilder _indexRebuilder;
+    private readonly IAppPolicyCache _runtimeCache;
+
+    public RebuildIndexExamineManagementController(
+        IExamineManagerService examineManagerService,
+        ILogger<RebuildIndexExamineManagementController> logger,
+        AppCaches runtimeCache,
+        IIndexRebuilder indexRebuilder)
+    {
+        _examineManagerService = examineManagerService;
+        _logger = logger;
+        _indexRebuilder = indexRebuilder;
+        _runtimeCache = runtimeCache.RuntimeCache;
+    }
+
+    /// <summary>
+    ///     Rebuilds the index
+    /// </summary>
+    /// <param name="indexName"></param>
+    /// <returns></returns>
+    [HttpPost("rebuildIndex")]
+    [MapToApiVersion("1.0")]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(OkResult), StatusCodes.Status200OK)]
+    public async Task<IActionResult> RebuildIndex(string indexName)
+    {
+        if (!_examineManagerService.ValidateIndex(indexName, out IIndex? index))
+        {
+            var invalidModelProblem = new ProblemDetails
+            {
+                Title = "Index Not Found",
+                Detail = $"No index found with name = {indexName}",
+                Status = StatusCodes.Status400BadRequest,
+                Type = "Error",
+            };
+
+            return BadRequest(invalidModelProblem);
+        }
+
+        if (!_examineManagerService.ValidatePopulator(index!))
+        {
+            var invalidModelProblem = new ProblemDetails
+            {
+                Title = "Index cannot be rebuilt",
+                Detail = $"The index {index?.Name} cannot be rebuilt because it does not have an associated {typeof(IIndexPopulator)}",
+                Status = StatusCodes.Status400BadRequest,
+                Type = "Error",
+            };
+
+            return BadRequest(invalidModelProblem);
+        }
+
+        _logger.LogInformation("Rebuilding index '{IndexName}'", indexName);
+
+        // remove it in case there's a handler there already
+        index!.IndexOperationComplete -= Indexer_IndexOperationComplete;
+
+        //now add a single handler
+        index.IndexOperationComplete += Indexer_IndexOperationComplete;
+
+        try
+        {
+            var cacheKey = "temp_indexing_op_" + index.Name;
```

Comment:
It would be nice with a small service that could handle setting, clearing and detecting `temp_indexing_op_` from the runtime cache ... it could then eventually be turned into something less rudimentary without breaking the examine management implementation.